### ISSUE TITLE
Redirect Request for Find a Representative (Accredited Representation Management)

### ISSUE THREAD

**Issue by oddball-lindsay** [department-of-veterans-affairs/va.gov-team#69952]
### Status

**[2023-03-18]** Lindsay updated that the ARM PO would now like to move forward with the third redirect ...but then quickly pulled back.  No changes, only proceeding with the first two redirects.

**[2023-03-15]** Confirmation from Lindsay that only the first two URLs will be released; she crossed out the one that won't be released.
**[2024-02-28]** Moving forward with a 3/18 push, updates have been made in the table below.

**[2024-03-05 Fran]** Lindsey commented: We have found a solution to our launch-blocking issue. We hope to verify the fix tomorrow, which means we could be ready to being our staged rollout either late 3/6 or early 3/7. This means that our **redirect date will push to 3/19 or 3/20.** I'll know more tomorrow afternoon and will report back!

**[2024-02-28]** Per Kristin, a go/no go decision will be made week of 3/5.

### Type of request
- [x] We are retiring or taking down a page and need to redirect the URL (complete redirect section)
- [x] We are changing the URL of an existing page (complete redirect section)

### Implementation date
When does this request need to be live?

Choose one:
- [x] On a specific date, connected to a program or publishing requirement. See dates below. [Staged Rollout](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/accredited-representation-management/product-documentation/find-a-representative/release-plan-find-a-representative-mvp.md#staged-rollout-details)

Additional context and timing in the [Redirects and Entry Point Plan
](https://docs.google.com/document/d/1Wxue2bQWHXY2fUGTTvRE-WKOUSJkUvUN5avpCooImc0/edit)

### Redirects 
| Status                          | Current URL                                                       | Redirect To                                     | Date/timing of redirect                                                                                                           |
|---------------------------------|-------------------------------------------------------------------|-------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|
| Stakeholders approved | https://www.va.gov/decision-reviews/get-help-with-review-request/ | va.gov/get-help-from-accredited-representative/ | 3/18/2024 |
| Stakeholders approved | https://www.va.gov/disability/get-help-filing-claim/              | va.gov/get-help-from-accredited-representative/ | 3/18/2024 |
| ~Stakeholders approved redirect to FAQ, not this landing page. ARM PO is not ready to move forward with this redirect~ | ~https://benefits.va.gov/vso/index.asp~                             | ~va.gov/get-help-from-accredited-representative~ | ~**Confirming**: 3/18/2024 if possible, otherwise open to other dates~              |

### Process, Roles and Responsibilities
- [x] Requesting team: Above information is provided
- [x] Requesting team: All appropriate VA stakeholders are notified as appropriate
- [X] IA: Request is vetted and documented and implementation plan is clear 
- [X] IA: Request is assigned to appropriate team for implementation 
- [x] Implementation team: Work is complete
- [x] Implementation team: Validated in production
- [x] Requesting team: Validates everything is correct in production and closes ticket

**Comment by oddball-lindsay** [comment]
@kristinoletmuskat here is our redirect ticket! A couple things to note:
1. I wasn't sure if we should mention the content changes (first through the React widget, then full copy change after we release to 100%)
2. Not sure if this type of request also counts as "We are retiring or taking down a page and need to redirect the URL (complete redirect section)" since we are doing that copy change
3. I don't think the Vanity URLs section applies to us, but wanted to double check before deleting it

**Comment by kristinoletmuskat** [comment]
Hi @jilladams and @FranECross, we're in the process of finalizing this redirect rollout plan, but wanted to tag ya'll on this one so you have it on your radar for your upcoming planning. When I spoke with Jill she said she would set up a time for us to sync on this plan with @aprocik1 from CAIA and @oddball-lindsay -- we're all new to this more nuanced redirect with a rollout toggle so wanted to make sure we're on the same page!

**Comment by jilladams** [comment]
Thanks @kristinoletmuskat -- our understanding of what is / isn't possible for Redirects is based on https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#redirect-types--examples. 

That in mind, looking at the URLs here, that means: 

1. 🟢 https://www.va.gov/decision-reviews/get-help-with-review-request/ - Case 1, modernized page, easy to redirect. Will go out with Wed devops deploy.
2. 🟢 https://www.va.gov/disability/get-help-filing-claim/ - Case 1, modernized page, easy to redirect. Will go out with Wed devops deploy.
3. 🟢 https://benefits.va.gov/vso/index.asp - Case 4a: Page-level cross-domain redirect for a site with injected header. We should be able to implement this, but it will not be a standard Devops deploy. Instead, this will be a client-side redirect in vets-website. That means: the user will initially land on the old URL, and then Javascript will load and bounce them to the new URL. This page will not redirect for users who have Javascript disabled. Also, if this page is ever archived in TeamSite in the future, that may affect whether or not the redirect successfully executes. More notes on that [here](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#4-page-level-cross-domain-redirects-from-subdomains-subdomainvagov-to-wwwvagov) under "Other notes" for your stakeholders to be aware of.
4. 🟢 / 🟡 https://www.va.gov/ogc/apps/accreditation/index.asp and all child pages - Probably Case 1, not modernized, but a va.gov domain, so may work fine with standard redirect, tbd. Sometimes pages that aren't modernized don't behave as expected with redirects, so we'll have to confirm once we can deploy.
5. 🟡 / 🔴 https://www.ebenefits.va.gov/ebenefits/vso-search --  This subdomain is not using the VA.gov injected header which means this is the 4b case, for Page-level cross-domain redirect without proxy-rewrite. I have yet to see a successful redirect get working for this case, but on paper: if it is a TeamSite, then the admins who own that TeamSite should be able to create the redirect. Are you engaged with stakeholders who own this site? If so, that may help us with getting it done.  If not, we may not be able to redirect this. (WebOps will not handle single page redirects for TeamSites; they require that the site admin manage it.)

**Comment by kristinoletmuskat** [comment]
@jilladams thank you for this thorough explanation! @oddball-lindsay would you be able to discuss Jill's point #5 with your stakeholder when you meet with them? Even if we don't want to redirect this page for this MVP, we will want to deal with this duplicate tool eventually,  so makes sense to get the conversations started now.

**Comment by jilladams** [comment]
Notes from talking to @kristinoletmuskat @oddball-lindsay @aprocik1 @randimays and @FranECross : 
* CAIA and AR team to finalize dates and list of redirects over the next week. 
* re: specific redirects: 
  * Public Websites will knock out the 1st 2 listed va.gov redirects by the listed deadline, when we get the greenlight.
  * The benefits.va.gov redirect will likely wait til Jan. 
  * * The ebenefits redirects are likely to move to another ticket, Kristin will decide / make that edit.
* We will keep this ticket in the Team repo so CAIA can continue to modify it as needed. PW will track it in sprint that way - no problem for us.

**Comment by kristinoletmuskat** [comment]
@jilladams I moved the eBenefits stuff out -- A and I are keeping [this google doc](https://docs.google.com/document/d/1Wxue2bQWHXY2fUGTTvRE-WKOUSJkUvUN5avpCooImc0/edit) up to date with the full redirect plan. I figured we could use this ticket for the 1/12/24 redirect, but if ya'll want I can create another ticket closer to that launch -- whatever works for you.

**Comment by oddball-lindsay** [comment]
@aprocik1 does this mean the only pending piece is the review for accuracy from the ARM stakeholders?

**Comment by aprocik1** [comment]
@oddball-lindsay, nope! Beth and I need to have the same conversation with our disability hub partners to redirect the second page in the table on this ticket. We've reached out and we're waiting on their response.

**Comment by chriskim2311** [comment]
@aprocik1 @oddball-lindsay @jilladams @FranECross 
First two redirects are ready to be merged when given okay from stakeholders.

**Comment by chriskim2311** [comment]
@aprocik1 @oddball-lindsay @jilladams @FranECross  Checking in on this ticket, if we would still like to make the 12/12 timeline. Please let me know if there's an update thank you!

**Comment by oddball-lindsay** [comment]
@chriskim2311 we had our "Go / No Go" meeting today and actually ended with "No Go" due to inaccuracies with the data powering our product's search results.  The current plan is to define a path to improve our data and reconvene tomorrow.  My gut is telling me we won't be ready by tomorrow 12/12, but I will know more by 3p ET on if we still want to proceed at a later date, potentially later this week.  I'll report back!

**Comment by oddball-lindsay** [comment]
@chriskim2311 we will not be ready to complete the redirects this week.  In fact, we are taking this sprint (12/12-12/25) to define a new MVP scope and release timeline.  When that timeline is ready, I'll be sure to communicate that with CAIA and we can update this ticket too.

**Comment by chriskim2311** [comment]
Thank you for the update! @oddball-lindsay 

@FranECross @jilladams Can we move this ticket to the backlog and pull in when we are ready?

**Comment by FranECross** [comment]
@chriskim2311 I'll move it into Blocked, pending a Go from Lindsay et al. Thanks! cc @jilladams

**Comment by kristinoletmuskat** [comment]
Hey @FranECross, we're planning t make he go/no go decision next week on 3/5.

**Comment by FranECross** [comment]
@chriskim2311 Just a head's up that you'll probably be tagged in the next week or so to release these redirects. Slack thread [here ](https://dsva.slack.com/archives/C05L6HSJLHM/p1709667754543699?thread_ts=1709662742.871449&cid=C05L6HSJLHM)for current convo. Thanks! cc @jilladams

**Comment by oddball-lindsay** [comment]
@FranECross @chriskim2311 @jilladams we have found a solution to our launch-blocking issue.  We hope to verify the fix tomorrow, which means we could be ready to being our staged rollout either late 3/6 or early 3/7.  

This means that our redirect date will push to 3/19 or 3/20 I''ll know more tomorrow afternoon and will report back!

**Comment by FranECross** [comment]
Thanks for the update, @oddball-lindsay!!

**Comment by oddball-lindsay** [comment]
Remaining close here -- things are looking good and we're anticipating deployment of our fix this afternoon, but it's in the team's best interest to fully review changes before heading into Staged Rollout.  

Assuming no surprises come up after deployment, Plan A is to confirm everything looks good by tomorrow morning and go live tomorrow afternoon.  If we do discovery any lingering issues, then Plan B is to go live on Monday (we are concerned about going live on a Friday and then missing any issues over the weekend).  Another "report back" coming tomorrow afternoon :)

**Comment by oddball-lindsay** [comment]
I'm back a bit sooner than "tomorrow afternoon" -- given the progress of review today, we think it's safest to not rush things and proceed with the Monday start date.  Here's our updated staged rollout schedule

1. Stage A: 10% of users on 3/11 (Monday)
2. Stage B: 25% of users on 3/13 (Wednesday)
3. Stage C: 50% of users on 3/18 (Monday)
4. Stage D: 100% of users on 3/20 (Wednesday) - monitor for any issues through 3/25 EOD (Monday)
   1. On 3/21 (Thursday) CAIA will implement the redirects and search result "best bets", as well as begin updating the entry points
   
I've updated the table up top to reflect that 3/21 redirect date!

@FranECross @chriskim2311 @jilladams @kristinoletmuskat

**Comment by FranECross** [comment]
@oddball-lindsay Confirming the 3/21 redirect date as noted above. 

@chriskim2311 @randimays We'll plan to pull this into the sprint starting this week (3/13). Please let me know if you have any concerns. Thanks!

**Comment by FranECross** [comment]
@chriskim2311 @randimays FYSA the new go-live for this item is 3/18. Thanks!

**Comment by chriskim2311** [comment]
@oddball-lindsay @FranECross  To clarify the table above. We are not going to update this URL yet https://benefits.va.gov/vso/index.asp correct? Just the first 2 for the 3/18 go-live date?

**Comment by oddball-lindsay** [comment]
@chriskim2311 I think we will move forward with the 3rd, but just double checking -- will confirm by EOD Friday so that you'll have clear direction come Monday

**Comment by oddball-lindsay** [comment]
@chriskim2311 our [PO has advised us not to move forward with that third redirect](https://dsva.slack.com/archives/C05L6HSJLHM/p1710519353162569?thread_ts=1710348844.663559&cid=C05L6HSJLHM), so I've made that update in the ticket summary up top.  Just the two redirects for Monday 3/18, please!

**Comment by chriskim2311** [comment]
Got it thanks @oddball-lindsay! Code will merge this afternoon for the redirect of the first two. They will go out Monday afternoon.

**Comment by oddball-lindsay** [comment]
@chriskim2311 @FranECross @randimays given all of our back and forth in this ticket, i'm embarrassed to come back and change our direction for the third redirect https://benefits.va.gov/vso/index.asp 😅  

We just had a meeting this morning and our PO is ready to move forward with redirecting to our landing page.  I've updated the table up top to reflect this.  If today is too short notice, happy to explore a different day for this redirect -- please let me know what works best given your schedules!

**Comment by chriskim2311** [comment]
@oddball-lindsay Thank you for the update. We have the PR ready to be merged when ready. Just let us know when a date has been finalized.

**Comment by oddball-lindsay** [comment]
@chriskim2311 to clarify - do you mean the PR for the https://benefits.va.gov/vso/index.asp redirect is ready to be merged whenever?

**Comment by chriskim2311** [comment]
@oddball-lindsay Yup! I have it approved and ready to go when given the go ahead.

**Comment by oddball-lindsay** [comment]
ok thanks!  i'll likely open a new ticket for that redirect when we're ready :)

**Comment by chriskim2311** [comment]
@oddball-lindsay Redirects for the first 2 are live on prod!

**Comment by oddball-lindsay** [comment]
looks great 💯 thanks for your help @chriskim2311!
